lib/Core/Inference/SymbolContextResolver.php

```diff
@@ -457,7 +457,18 @@ private function resolveParameterFromReflection(Frame $frame, MethodDeclaration
 
     private function resolveNumericLiteral(NumericLiteral $node): SymbolContext
     {
-        $value = eval(sprintf('return %s;', $node->getText()));
+        $value = $node->getText();
+        if (0 === strpos($value, '0b')) {
+            $value = bindec(substr($value, 2));
+        } elseif (0 === strpos($value, '0x')) {
+            $value = hexdec(substr($value, 2));
+        } elseif (1 === preg_match('/^0\\d+$/', $value)) {
+            $value = octdec(substr($value, 1));
+        } elseif (1 === preg_match('/^\\d+\\.\\d*$/', $value)) {
+            $value = (float) $value;
+        } else {
+            $value = (int) $value;
+        }
```

Comment:
in general i try and avoid `else` in Phpactor, you could do this without assigns by delegating to another method, or using a closure:

```
$value = (function ($value) {
    if ($value == 'blah') {
        return $value;
    }
    // ... etc
    return (int) $value;
})($value);
```

Reply:
May I ask you what the benefits would be?
I guess it would just be slower, wouldn't it?

Reply:
it's a general coding convention:

- preferring early returns and guard conditions eliminates the need for else
- `else` often causes methods that start with `if` and 200 lines later you have no idea if there is an `else` and what the return value would be or if that value will be mutated by something.
- removing multiple assignments, above there is only one assignment.
- encapsulation...

it is a bit slower

```

    benchIf.................................I9 ✔ Mo0.809μs (±26.12%)
    benchClosure............................I9 ✔ Mo1.517μs (±10.56%)
```